scripts/ock-setup.sh

```diff
@@ -0,0 +1,262 @@
+title ENDSECTION
+
+title SECTION "Setup"
+
+title PARA "Directories and files"
+title LINE "reset temporary directory"
+rm -rf ${TMPDIR}
+mkdir -p ${TMPDIR}
+
+echo ${PIN} > ${PINFILE}
+
+R64K=${TMPDIR}/r64k.bin
+dd if=/dev/urandom of=${R64K} bs=64K count=1 &> /dev/null
+
+R256=${TMPDIR}/r256.bin
+dd if=/dev/urandom of=${R256} bs=32 count=1 &>/dev/null
+
+R512=${TMPDIR}/r512.bin
+dd if=/dev/urandom of=${R512} bs=64 count=1 &>/dev/null
+
+title PARA "Openssl configuration (prepare)"
+if [ -z "${OPENSSL_CONF}" ]; then
+    title LINE "skip (OPENSSL_CONF must be set)"
+    title ENDSECTION
+    exit 77
+fi
+if [ ! -f "${OPENSSL_CONF}" ]; then
+    title LINE "skip (${OPENSSL_CONF} missing)"
+    title ENDSECTION
+    exit 77
+fi
+sed -e "s,^pkcs11-module-path = .*$,pkcs11-module-path = ${LIBOCK}," \
+    -e "/^pkcs11-module-init-args = .*$/d" \
+    -e "s,^pkcs11-module-token-pin = .*$,pkcs11-module-token-pin = file:${PINFILE}," \
+    ${OPENSSL_CONF} \
+> ${TMPDIR}/openssl.cnf
+unset OPENSSL_CONF
+title LINE "OPENSSL_CONF ${OPENSSL_CONF}"
+
+title PARA "generate test keys (rsa)"
+
+LBL="ock_rsa2k"
+P11SAK_ARGS="--slot ${SLOT} --pin ${PIN}"
+
+title LINE "clean-up"
+${P11SAK} remove-key rsa --label "${LBL}:prv" --force ${P11SAK_ARGS} > /dev/null || true
+${P11SAK} remove-key rsa --label "${LBL}:pub" --force ${P11SAK_ARGS} > /dev/null || true
+
+title LINE "generate key-pair"
+${P11SAK} generate-key rsa 2048 --attr "X:PSx" --label ${LBL} ${P11SAK_ARGS}
+
+title PARA "RSA PKCS11 URIs"
+BASEFILE="${TMPDIR}/${LBL}"
+BASEURI=""
+BASEURIWITHPIN=""
+PUBURI="pkcs11:object=${LBL}:pub;type=public"
+PRIURI="pkcs11:object=${LBL}:prv;type=private"
+title LINE "BASEFILE       ${BASEFILE}"
+title LINE "BASEURIWITHPIN ${BASEURIWITHPIN}"
+title LINE "BASEURI        ${BASEURI}"
+title LINE "PUBURI         ${PUBURI}"
+title LINE "PRIURI         ${PRIURI}"
+
+title PARA "generate test keys (ec)"
+LBL="ock_ec256"
+
+title LINE "clean-up"
+${P11SAK} remove-key ec --label "${LBL}:prv" --force ${P11SAK_ARGS} > /dev/null || true
+${P11SAK} remove-key ec --label "${LBL}:pub" --force ${P11SAK_ARGS} > /dev/null || true
+
+title LINE "generate key-pair"
+${P11SAK} generate-key ec prime256v1 --attr "X:PSx" --label ${LBL} ${P11SAK_ARGS}
+
+ECBASEFILE="${TMPDIR}/${LBL}"
+ECBASEURI=""
+ECPUBURI="pkcs11:object=${LBL}:pub;type=public"
+ECPRIURI="pkcs11:object=${LBL}:prv;type=private"
+
+title PARA "EC PKCS11 URIs"
+title LINE "ECBASEFILE      ${ECBASEFILE}"
+title LINE "ECBASEURI       ${ECBASEURI}"
+title LINE "ECPUBURI        ${ECPUBURI}"
+title LINE "ECPRIURI        ${ECPRIURI}"
+
+title PARA "Show contents of token in slot ${SLOT}"
+p11sak list-key all ${P11SAK_ARGS} | tail -n +2
+title LINE ""
+
+title PARA "openssl configuration (activate)"
+export OPENSSL_CONF="${TMPDIR}/openssl.cnf"
+title LINE "OPENSSL_CONF ${OPENSSL_CONF}"
+
+title PARA "Debug"
+title LINE "Export variables to ${TMPDIR}/debugvars for easy debugging"
+cat > ${TMPDIR}/debugvars << DBGSCRIPT
```

Comment:
this will conflict with nss-softoken, so we should probbaly use a different file like ock-debugvars?

This file is used in the shell to do something like:
source tests/tmp/ock-debugvars

and the idea is that I can simply run the last command that failed in the log and all works because all the required env vars are set in the shell.

Reply:
I did the separation on directory level, by using another tmp directory for opencryptoki (tmp-ock) and use the same file names. So I can switch between NSS and opencryptoki by source either `tmp/debugvars` (nss) or `tmp-ock/debugvars` (opencryptoki).